File: 1114-binary-search-tree-to-greater-sum-tree/1114-binary-search-tree-to-greater-sum-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorder(self, sum, root: Optional[TreeNode]):
        if not root:
            return
        self.postorder(sum, root.right)
        root.val += sum[0]
        sum[0] = root.val
        self.postorder(sum, root.left)

    def bstToGst(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        sum = [0]
        self.postorder(sum, root)
        return root

File: 1114-binary-search-tree-to-greater-sum-tree/1114-binary-search-tree-to-greater-sum-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    int sum = 0;
public:
    void rinorder(TreeNode* root) {
        if(root == NULL) return;
        rinorder(root->right);
        sum += root->val;
        root->val = sum;
        rinorder(root->left);
    }
    TreeNode* bstToGst(TreeNode* root) {
        rinorder(root);
        return root;
    }
};